tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -125,10 +126,30 @@ type storeImpl struct {
 func New(db postgres.DB) Store {
     return &storeImpl{
         db: db,
+        GenericStore: pgSearch.NewGenericStore{{ if .PermissionChecker }}WithPermissionChecker{{ end }}[{{.Type}}, *{{.Type}}](
+            db,
+            schema,
+            pkGetter,
+            metricsSetAcquireDBConnDuration,
+            metricsSetPostgresOperationDurationTime,
+            {{ if .PermissionChecker }}{{ .PermissionChecker }}{{ else }}targetResource{{ end }},
+        ),
     }
 }
 
-//// Helper functions
+// region Helper functions
+
+func pkGetter(obj *{{ .Type }}) {{$singlePK.Type}} {
+    return {{ $singlePK.Getter "obj" }}
+}
+
+func metricsSetPostgresOperationDurationTime(start time.Time, op ops.Op) {
+    metrics.SetPostgresOperationDurationTime(start, op, "{{.TrimmedType}}")
+}
+
+func metricsSetAcquireDBConnDuration(start time.Time, op ops.Op) {
+    metrics.SetAcquireDBConnDuration(start, op, "{{.TrimmedType}}")
```

Comment:
Would a constant extraction make any sense?
```go
const storeName = {{.TrimmedType | quote}}
```
And what about type aliase?
```go
type storeType = {{.Type}}
```

Reply:
Reducing templated code, as with the string constant. That will allow us to extract non-templates from `.tpl`.

Reply:
Addressed in 
- https://github.com/stackrox/stackrox/pull/6778